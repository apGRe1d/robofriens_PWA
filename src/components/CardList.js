import React from 'react';
import Card from './Card';

const CardList = ({ robots }) => {
  // throw new Error('ERRROOOORRR');
  return (
    <div>
      {
        robots.map((user, i) => {
          return (
            <Card
              key={i}
              id={robots[i].id}
              name={robots[i].name}
              email={robots[i].email}
              />
          );
        })
      }
    </div>
  );
}

CardList.whyDidYouRender = true;
export default CardList;
