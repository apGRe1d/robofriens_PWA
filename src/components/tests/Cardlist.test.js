import { shallow, mount, render } from 'enzyme';
import CardList from '../CardList';
import React from 'react';

it('expect to render CardList component ', function () {
  const mockRobots = [
    {
      key: 1,
      id: 1,
      name: 'John John',
      email: 'JohnJohn@gmail.com'
    }
  ]
  expect(shallow(<CardList robots={mockRobots} />)).toMatchSnapshot();
});
