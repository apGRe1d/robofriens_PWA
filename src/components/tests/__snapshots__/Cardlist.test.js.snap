// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CardList component  1`] = `
<div>
  <Card
    email="JohnJohn@gmail.com"
    id={1}
    key="0"
    name="John John"
  />
</div>
`;
